sound/soc/sof/nocodec.c

```diff
@@ -42,12 +42,7 @@ int sof_nocodec_setup(struct device *dev,
 
 	ret = sof_bes_setup(dev, ops, links, ops->num_drv,
 			    &sof_nocodec_card);
-	if (ret) {
-		kfree(links);
```

Comment:
I believe Coccinelle will not complain if we change it to 'devm_kfree(dev, links);'
But I like reporting error and stopping probing, leave the devm free to the device freeing itself.